board/drivers/fdcan.h

```diff
@@ -112,7 +112,7 @@ void process_can(uint8_t can_number) {
           fifo = (canfd_fifo *)(TxFIFOSA + (tx_index * FDCAN_TX_FIFO_EL_SIZE));
 
           fifo->header[0] = (to_send.extended << 30) | ((to_send.extended != 0U) ? (to_send.addr) : (to_send.addr << 18));
-          fifo->header[1] = (to_send.data_len_code << 16) | (bus_config[can_number].canfd_enabled << 21) | (bus_config[can_number].brs_enabled << 20);
+          fifo->header[1] = (typeof(fifo->header[1]))(to_send.data_len_code << 16) | (bus_config[can_number].canfd_enabled << 21) | (bus_config[can_number].brs_enabled << 20);
```

Comment:
why does this line need a typeof and the others like it don't? I suspect there's a cleaner fix here

Reply:
hmm... I changed the canfd_enabled to uint32_t since it is shifted and updated to be of int type anyway. I don't really like changing it to another type like this though, may be do inline suppress for this?

Reply:
I think the way cppcheck catch this is looking at the type of canfd_enabled and data_len_code, and it doesn't take into account that bool type get shifted turn into an int.